control-plane-operator/controllers/hostedcontrolplane/v2/assets/ingress-operator/deployment.yaml

```diff
@@ -0,0 +1,97 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ingress-operator
+spec:
+  replicas: 1
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      name: ingress-operator
+  strategy:
+    type: Recreate
```

Comment:
why is this set here? should we enforce this value?

Reply:
not all components use `Recreate`, a lot of them use `RollingUpdate`

Reply:
I think we should enable this via library .withStrategy

Reply:
I think it makes sense to leave it in the manifest to reduce the code part and keep the manifest as the authoritative source.
We also support statefulsets and cronJobs (soon via https://github.com/openshift/hypershift/pull/5663) which don't use the same field type,